### PR TITLE
DEVOP-1065: Add scripts to setup instance

### PR DESCRIPTION
**Pull request by sboardwell**

### CODE REVIEW

File: download.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    mkdir -p "${installPath}"
+    curl -fsSL -o "${installPath}/${downloadFile}" "${downloadUrl}"
+    cd "${installPath}"
+    info "Extracting '${downloadFile}'"
+    tar xzf "${downloadFile}" --strip 1
+    cd -
+    info "Setting symlink '${installPath}' -> '${latestPath}'"
+    rm -f "${latestPath}"
+    ln -sf "${installPath}" "${latestPath}"
+  fi
```
Comment (sboardwell) [inline download.sh:71]: Add a echo as to where the files have been downloaded - optionally with the command to continue.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+source "${scriptsDir}/functions.sh"
+
+checkVars() {
+  if [ -z "${INSTALL_USER:-}" ]; then
+    prompt INSTALL_USER "What is your Linux Username? Leave blank to use the current user." "$(whoami)"
+  else
+    echo "Found INSTALL_USER = '$INSTALL_USER'"
+  fi
+  if [ -z "${INSTALL_USER_HOME:-}" ]; then
+    prompt INSTALL_USER_HOME "Where is your Linux Home Directory? Leave blank to use the current user's Home." "$(eval echo ~$INSTALL_USER)"
```
Comment (sboardwell) [inline setup.sh]:
```suggestion
    prompt INSTALL_USER_HOME "Where is your Linux Home Directory? Leave blank to use the current user's Home." "$(getent passwd "$INSTALL_USER" | cut -d: -f6)"
```

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+set -euo pipefail
+
+currentScriptDir="$( cd "$( dirname "${BASH_SOURCE[0]:-}" )" >/dev/null 2>&1 && pwd )"
+scriptsDir="${currentScriptDir}/scripts"
+source "${scriptsDir}/functions.sh"
+
+checkVars() {
+  if [ -z "${INSTALL_USER:-}" ]; then
+    prompt INSTALL_USER "What is your Linux Username? Leave blank to use the current user." "$(whoami)"
```
Comment (sboardwell) [inline setup.sh]:
```suggestion
    prompt INSTALL_USER "Which user will use the installation (e.g. ec2-user)? Leave blank to use the current user." "$(whoami)"
```

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  else
+    echo "Found INSTALL_USER_HOME = '$INSTALL_USER_HOME'"
+  fi
+  if [ -z "${PLATFORM_ECR_IMAGE:-}" ]; then
+    prompt PLATFORM_ECR_IMAGE "What is the Platform ECR Image URI?" ""
+  else
+    echo "Found PLATFORM_ECR_IMAGE = '$PLATFORM_ECR_IMAGE'"
+  fi
+  if [ -z "${AWS_ECR_CREDENTIALS:-}" ]; then
+    prompt AWS_ECR_CREDENTIALS "What are your AWS ECR Credentials? Input in the form: <AWS_ACCESS_KEY_ID>:<AWS_SECRET_ACCESS_KEY>" ""
```
Comment (sboardwell) [inline setup.sh:26]: Add read as secret

Comment (sboardwell) [reply setup.sh:26]: won't do

**Review by sboardwell — COMMENTED** [review]
lgtm

**Review by shishir-narayan — APPROVED** [review]
lgtm

**Pull request merged**